File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/domain/medal/MedalAcquisitionConditionCreator.java
package com.depromeet.threedollar.domain.domain.medal;

import com.depromeet.threedollar.common.docs.ObjectMother;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
class MedalAcquisitionConditionCreator {

    public static MedalAcquisitionCondition create(Medal medal, MedalAcquisitionConditionType conditionType, int count, String description) {
        return MedalAcquisitionCondition.builder()
            .medal(medal)
            .conditionType(conditionType)
            .count(count)
            .description(description)
            .build();
    }

}

File: threedollar-domain/src/testFixtures/java/com/depromeet/threedollar/domain/user/domain/medal/MedalAcquisitionConditionCreator.kt
package com.depromeet.threedollar.domain.user.domain.medal

import com.depromeet.threedollar.domain.user.domain.ObjectMother

@ObjectMother
object MedalAcquisitionConditionCreator {

    @JvmStatic
    fun create(
        medal: Medal,
        conditionType: MedalAcquisitionConditionType,
        count: Int,
        description: String
    ): MedalAcquisitionCondition {
        return MedalAcquisitionCondition.builder()
            .medal(medal)
            .conditionType(conditionType)
            .count(count)
            .description(description)
            .build()
    }

}
